runtime-modules/content/src/benchmarks/benchmarking.rs

```diff
@@ -290,6 +291,511 @@ benchmarks! {
+
+    }: _ (origin,
+        actor,
+        channel_id,
+        storage_buckets_num_witness,
+        a.into(),
+        rationale.clone())
+    verify {
+        assert_last_event::<T>(
+            Event::<T>::ChannelDeletedByModerator(
+                actor,
+                channel_id,
+                rationale
+            ).into()
+        );
+    }
+
+    set_channel_visibility_as_moderator{
+
+        let a in 1 .. MAX_BYTES_METADATA; //max bytes for rationale
+
+        let (
+            channel_id,
+            group_id,
+            _,
+            curator_id,
+            curator_account_id
+        ) =
+        setup_worst_case_scenario_curator_channel::<T>(
+            T::MaxNumberOfAssetsPerChannel::get(),
+            T::StorageBucketsPerBagValueConstraint::get().max() as u32,
+            T::DistributionBucketsPerBagValueConstraint::get().max() as u32,
+            true
+        ).unwrap();
+
+        let origin = RawOrigin::Signed(curator_account_id);
+        let actor = ContentActor::Curator(group_id, curator_id);
+
+        let rationale = vec![0u8].repeat(a as usize);
+
+    }: _ (origin, actor, channel_id, true, rationale.clone())
+    verify {
+
+        assert_last_event::<T>(
+            Event::<T>::ChannelVisibilitySetByModerator(
+                actor,
+                channel_id,
+                true,
+                rationale,
+            ).into());
+    }
+
+    delete_video_assets_as_moderator_with_assets {
+
+        let a in 1 .. T::MaxNumberOfAssetsPerVideo::get(); //max objs number
+
+        let b in (T::StorageBucketsPerBagValueConstraint::get().min as u32) ..
+         (T::StorageBucketsPerBagValueConstraint::get().max() as u32);
+
+        let c in 1 .. MAX_BYTES_METADATA; //max bytes for rationale
+
+        let (
+            channel_id,
+            group_id,
+            _,
+            curator_id,
+            curator_account_id
+        ) =
+        setup_worst_case_scenario_curator_channel::<T>(
+            T::MaxNumberOfAssetsPerChannel::get(),
+            b,
+            T::DistributionBucketsPerBagValueConstraint::get().max() as u32,
+            false
+        ).unwrap();
+
+        let origin = RawOrigin::Signed(curator_account_id);
+        let actor = ContentActor::Curator(group_id, curator_id);
+
+        let rationale = vec![1u8].repeat(c as usize);
+        let storage_buckets_num_witness =
+            Some(storage_buckets_num_witness::<T>(channel_id)?);
+
+        let assets_to_remove_start = T::MaxNumberOfAssetsPerChannel::get();
+        let assets_to_remove_end = T::MaxNumberOfAssetsPerChannel::get() + a;
+
+        let assets_to_remove: BTreeSet<T::DataObjectId> =
+            (assets_to_remove_start..assets_to_remove_end)
+                .map(|i| i.saturated_into()).collect();
+
+        let video_id = setup_video::<T>(
+            origin.clone().into(),
+            actor,
+            channel_id,
+            a,
+            true,
+            MAX_BYTES_METADATA);
```

Comment:
You should use a worst case scenario video nft in this case, which is an nft in `TransactionalStatus::EnglishAuction` with a `whitelist` of maximum possible length.

Because `MaxAuctionWhiteListLength` is a storage value, this means we need to introduce additional `Config` value which defines a max value for this storage value and then use this `Config` value in the benchmarks.